From eefc722e664fa19d5ba9f23cf777d858eae9370e Mon Sep 17 00:00:00 2001
From: Jonas Larsson <ljonas@google.com>
Date: Mon, 26 Aug 2019 14:19:04 -0700
Subject: [PATCH 06/17] Fix regression of BGRA upload color channel switched

Commit 300b01b0a8b11a266f0cdd6233750fafc1e2125c ported the fix
from plugins-bad commit a4b72743dd5b8ead623f4ce92a1b05ef791db67f
but added the call too _drm_direct_fourcc_from_info to the wrong
function.

Change-Id: I6051ec4f501cdde234e1503c5f0d569135f793d5
---
 gst-libs/gst/gl/egl/gsteglimage.c | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/gst-libs/gst/gl/egl/gsteglimage.c b/gst-libs/gst/gl/egl/gsteglimage.c
index 1ea478441..d020aa67b 100644
--- a/gst-libs/gst/gl/egl/gsteglimage.c
+++ b/gst-libs/gst/gl/egl/gsteglimage.c
@@ -421,7 +421,7 @@ _drm_fourcc_from_info (GstVideoInfo * info, int plane)
  * handle YUV formats directly (by using internal shaders, or hardwired
  * YUV->RGB conversion matrices etc.)
  */
-  static int
+static int
 _drm_direct_fourcc_from_info (GstVideoInfo * info)
 {
   GstVideoFormat format = GST_VIDEO_INFO_FORMAT (info);
@@ -537,10 +537,7 @@ gst_egl_image_from_dmabuf (GstGLContext * context,
   gint fourcc;
   gint i;
 
-  fourcc = _drm_direct_fourcc_from_info (in_info);
-  if (fourcc == -1)
-    return NULL;
-
+  fourcc = _drm_fourcc_from_info (in_info, plane);
   format = gst_gl_format_from_video_info (context, in_info, plane);
 
   GST_DEBUG ("fourcc %.4s (%d) plane %d (%dx%d)",
@@ -652,7 +649,9 @@ gst_egl_image_from_dmabuf_singleplaner (GstGLContext * context,
   };
   EGLImageKHR img = EGL_NO_IMAGE_KHR;
 
-  fourcc = _drm_fourcc_from_video_info (in_info);
+  fourcc = _drm_direct_fourcc_from_info (in_info);
+  if (fourcc == -1)
+    return NULL;
 
   GST_DEBUG ("fourcc %.4s (%d) n_planes %d (%dx%d)",
       (char *) &fourcc, fourcc, n_planes,
-- 
2.26.0

